docs/proposals/20210628-nodepool-ingress-support.md

```diff
@@ -0,0 +1,230 @@
+	      and add specific filter operation for it
+
+	Conclusion:
+		By evaluating all the alternatives above, we think Yurthub naturally can take the responsibility to filter the
+		data between Cloud and Edge, so we prefer to Solution 4 currently, thanks @wenjun93 for his great suggestions
+		about this proposal, any other suggestions will also be appreciated, if no different opinions, we will follow
+		Solution 4 to implement the feature.
+
+### User Stories
+
+#### Story 1
+As an end user, I want to access edge services in a NodePool through http/https.
+#### Story 2
+As an end user, I want to develop an app which needs to utilize multi services in a NodePool.
+#### Story 3
+As an end user, I want to balance the traffic into a NodePool with layer-7 load-balancer mechanism and bypass
+layer-4 kube-proxy.
+
+### Implementation Details/Notes/Constraints
+1). Yurt-app-manager:
+	- When a NodePool is created, it can be labeled whether to enable the ingress feature, for example:
+	  ```yaml
+            nodepool.openyurt.io/ingress-enable: "true"
+	  ```
+	  Whether to enable ingress by default when NodePool created is TBD.
+	- A new CRD will be defined to provide the daemonset support from NodePool level, so that one ingress controller
+	  can be deployed to the NodePool which needs to enable ingress feature.
+	- For every ingress controller deployment in different NodePools, it creates a NodePort type service to expose the
+	  ingress controller service, and binds a corresponding configmap to differentiate the NodePools. All the ingress
+	  controller deployments for different NodePools can use one same service account.
+	- Add 2 NodePool annotations for users to configure the ingress feature when NodePool is created:
+	  ```yaml
+            nodepool.openyurt.io/ingress-ips: "xxx"      # the ingress controller service IP scope
+            nodepool.openyurt.io/ingress-config: "xxx"   # the ingress controller nginx config
+	  ```
+
+2). Nginx ingress controller:
+	- Keep the upstream version none-intrusive, so that it will be very convenient to upgrade in future.
+	- It specifies the ingress class to NodePool name so the ingress CR can select the corresponding ingress controller.
+	- It should be granted the authorities to create/update/patch the related resources in kube-system namespace.
+	- It connects with Yurthub to monitor the resources(ingress/service/configmap/secret/endpoint) of its own NodePool.
+
+3). Ingress controller sidecar:
+	- When sidecar is launched, it gets the NodePool id it belongs to, and connects to Yurthub to list/watch the NodePool.
+	- When it detects the NodePool annotation below is changed, it will update externalIPs of the corresponding service:
+	  ```yaml
+            nodepool.openyurt.io/ingress-ips: "xxx"
+	  ```
+	- When it detects the NodePool annotation below is changed, it will update the corresponding configmap of the NodePool,
```

Comment:
If the workloads in NodePool use different configmaps, NodePool daemonset resource need to support different configmap setting. so how about create different configmaps directly by user instead of setting NodePool ingress-config?

Reply:
About the details design for implementation, it's just a draft version by now.  We will investigate more to make it clear.
Thanks for your suggestion, we will take it into account.